src/main/java/seedu/cafectrl/data/dish/Ingredient.java

```diff
@@ -38,8 +38,16 @@ public String toString() {
         return name + " - " + qty + unit;
     }
 
+    /**
+     * Compare the name of this ingredient to the other ingredient
+     * @param obj the other ingredient to be compared with
+     * @return true if they have the same name, false otherwise
+     */
     @Override
     public boolean equals(Object obj) {
-        return this.toString().equals(obj.toString());
+        if (obj instanceof Ingredient) {
+            return this.name.equals(((Ingredient) obj).name);
```

Comment:
perhaps you want to create variable for `(Ingredient) obj).name` to simplify this statement